### PR TITLE
Add Arm64EC to inline-assembly documentation

### PR DESCRIPTION
**Pull request by taiki-e**
This PR is a sub-part of https://github.com/rust-lang/rust/pull/131781.

cc @Amanieu

@rustbot label +S-waiting-on-stabilization

### CODE REVIEW

**Comment by ehuss** [comment]
@Amanieu Can you please review for correctness?

**Review by Amanieu — COMMENTED** [review]
LGTM with regards to correctness. Just a nit to avoid unnecessary duplication.

**Review by Amanieu — APPROVED** [review]

File: src/inline-assembly.md [hunk 1]
```diff
@@ -432,6 +446,16 @@ The supported modifiers are a subset of LLVM's (and GCC's) [asm template argumen
...
+| Arm64EC | `reg` | None | `x0` | `x` |
+| Arm64EC | `reg` | `w` | `w0` | `w` |
+| Arm64EC | `reg` | `x` | `x0` | `x` |
+| Arm64EC | `vreg` | None | `v0` | None |
+| Arm64EC | `vreg` | `v` | `v0` | None |
+| Arm64EC | `vreg` | `b` | `b0` | `b` |
+| Arm64EC | `vreg` | `h` | `h0` | `h` |
+| Arm64EC | `vreg` | `s` | `s0` | `s` |
+| Arm64EC | `vreg` | `d` | `d0` | `d` |
+| Arm64EC | `vreg` | `q` | `q0` | `q` |
```
[the remark this answers is not in the document]
Comment (taiki-e) [reply src/inline-assembly.md]: That makes sense. Addressed.

**Review by Amanieu — APPROVED** [review]

**Comment by taiki-e** [comment]
https://github.com/rust-lang/rust/pull/131781 has been merged and rebased to resolve merge conflict with https://github.com/rust-lang/reference/pull/1643.

@rustbot label -S-waiting-on-stabilization

**Review by ehuss — APPROVED** [review]
Thanks!

**Pull request merged**